examples/utils/generic.py

```diff
@@ -91,27 +139,27 @@ def ensure_packages_are_installed(*names):
     if len(names) > 0:
         for name in names:
             if importlib.util.find_spec(name) is None:
-                install_package(name)
+                install_package(name, notebook_environment)
 
     # Install requirements.txt if nothing was passed in
     else:
-        reqs_file = os.path.realpath(os.path.join(__file__, "../../../requirements.txt"))
+        if notebook_environment == "Colab":
+            reqs_file = os.path.realpath(os.path.join(__file__, "../../../requirements-colab.txt"))
+        else:
+            reqs_file = os.path.realpath(os.path.join(__file__, "../../../requirements.txt"))
         with open(reqs_file, "r") as reqs:
             for line in reqs:
                 # Ignore Jupyterlab, since the user is probably running it already to view the notebooks
                 if "jupyterlab" in line:
                     pass
                 # Check if packages exist, and install if they don't
                 else:
-                    name, version = line.strip().split("==")
-                    if importlib.util.find_spec(name) is None:
-                        install_package(name, version)
-
-
-ensure_packages_are_installed("tabulate")
-
-# Needs to go here, to ensure it's installed before we import
-from tabulate import tabulate
+                    # If we want to add comments to requirements.txt files, we must
+                    # consider only lines in the file with '=='
```

Comment:
What is a comment contains "=="?

```py
# This is a comment with == in it
```

A better logic would be to have something like:

```py
if line.startswith("#"): pass
```